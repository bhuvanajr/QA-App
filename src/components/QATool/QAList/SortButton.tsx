import React from 'react';
import { useSortContext } from 'context';

export const SortButton: React.FC = () => {
  const {
    methods: { requestSort, getSortIconClassName },
  } = useSortContext();
  const iconClass = getSortIconClassName('question');

  return (
    <button type="button" className="btn-icon" onClick={() => requestSort('question')}>
      {iconClass && <div className={`icon ${iconClass}`} />}
    </button>
  );
};
