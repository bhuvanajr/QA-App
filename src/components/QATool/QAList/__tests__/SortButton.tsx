import { render, screen } from '@testing-library/react';
import React from 'react';
import { SortButton } from '../SortButton';
import userEvent from '@testing-library/user-event';

const mockrequestsort = jest.fn();
const mockgetsorticonclassName = jest.fn();

jest.mock('context', () => ({
  useSortContext: () => ({
    methods: {
      requestSort: mockrequestsort,
      getSortIconClassName: mockgetsorticonclassName,
    },
  }),
}));

test('should call requestSort fn correctly', () => {
  render(<SortButton />);
  userEvent.click(screen.getByRole('button'));
  expect(mockrequestsort).toHaveBeenCalled();
});

test('should render sort button in ascending correctly', () => {
  mockgetsorticonclassName.mockReturnValueOnce('ascending');
  const { container } = render(<SortButton />);
  expect(container).toMatchSnapshot();
});

test('should render sort button in descending correctly', () => {
  mockgetsorticonclassName.mockReturnValueOnce('descending');
  const { container } = render(<SortButton />);
  expect(container).toMatchSnapshot();
});
