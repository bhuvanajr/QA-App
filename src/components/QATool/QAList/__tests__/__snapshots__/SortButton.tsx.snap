// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render sort button in ascending correctly 1`] = `
<div>
  <button
    class="btn-icon"
    type="button"
  >
    <div
      class="icon ascending"
    />
  </button>
</div>
`;

exports[`should render sort button in descending correctly 1`] = `
<div>
  <button
    class="btn-icon"
    type="button"
  >
    <div
      class="icon descending"
    />
  </button>
</div>
`;
